src/fetch.rs

```diff
@@ -46,8 +48,21 @@ pub fn get_latest_dependency(crate_name: &str, flag_allow_prerelease: bool) -> R
 
         return Ok(Dependency::new(crate_name).set_version(&new_version));
     }
+    if crate_name.is_empty() {
+        return Err(ErrorKind::EmptyCrateName.into());
+    }
+
+    let registry_path = registry_path()?;
+    let registry_url = registry_url()?;
+    if !offline {
+        UPDATE.call_once(|| {
+            if let Err(error) = update_git_repo(&registry_path, &registry_url) {
```

Comment:
I'm thinking, maybe we should move this functionality outside of `get_latest_dependency` and just call it when executing `cargo add` or `cargo upgrade` but before `get_latest_dependency`. @killercup do you agree?

Reply:
done.